File: typescript/examples/05-agent-search.ts
/**
 * Example 05 — Agent-oriented search (case card + opinion text + good-law)
 * Language: TypeScript (Node.js)
 *
 * One call returns enough context for a tool-using model to reason.
 */
import { LawDiverClient } from "../src/client.js";

const client = new LawDiverClient();

const found = await client.search({
  query: "can a landlord withhold a deposit for ordinary wear and tear",
  jurisdiction: { type: "one_state_plus_federal", state: "FL" },
  limit: 5,
  include: { caseCard: true, opinionText: true, goodLawReport: true },
  opinionTextMaxChars: 10000,
});

console.log(`requestId: ${found.requestId}`);
console.log(`engines:`, (found.searchInfo as { enginesUsed?: string[] } | undefined)?.enginesUsed);
console.log("");

for (const r of found.results) {
  if (r.goodLaw?.negative) {
    console.warn(`⚠ BAD LAW · ${r.caseName} · ${r.goodLaw.status}`);
  } else if (r.goodLaw?.unknown) {
    console.warn(`? UNKNOWN treatment · ${r.caseName}`);
  }

  const summary =
    (r.caseCard as { summaryAi?: string } | null)?.summaryAi?.slice(0, 160) ??
    "(no case card yet)";
  console.log(`• ${r.caseName} — ${r.citation}`);
  console.log(`  summary: ${summary}`);
  if (r.opinion) {
    console.log(
      `  opinion: mode=${r.opinion.mode} chars=${r.opinion.charCount} truncated=${r.opinion.truncated}`,
    );
    console.log(`  excerpt: ${r.opinion.text.slice(0, 120).replace(/\s+/g, " ")}…`);
  }
  console.log("");
}

File: python/examples/05_agent_search.py
"""Example 05 — Agent-oriented search (Python)."""

from __future__ import annotations

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from lawdiver import LawDiverClient

with LawDiverClient() as client:
    found = client.search(
        query="can a landlord withhold a deposit for ordinary wear and tear",
        jurisdiction={"type": "one_state_plus_federal", "state": "FL"},
        limit=5,
        include={"caseCard": True, "opinionText": True, "goodLawReport": True},
        opinionTextMaxChars=10000,
    )

print(f"requestId: {found['requestId']}")
info = found.get("searchInfo") or {}
print(f"engines: {info.get('enginesUsed')}")
print()

for r in found["results"]:
    gl = r.get("goodLaw") or {}
    if gl.get("negative"):
        print(f"⚠ BAD LAW · {r.get('caseName')} · {gl.get('status')}")
    elif gl.get("unknown"):
        print(f"? UNKNOWN treatment · {r.get('caseName')}")

    card = r.get("caseCard") or {}
    summary = (card.get("summaryAi") or "(no case card yet)")[:160]
    print(f"• {r.get('caseName')} — {r.get('citation')}")
    print(f"  summary: {summary}")
    opinion = r.get("opinion")
    if opinion:
        text = " ".join((opinion.get("text") or "").split())
        print(
            f"  opinion: mode={opinion.get('mode')} chars={opinion.get('charCount')} "
            f"truncated={opinion.get('truncated')}"
        )
        print(f"  excerpt: {text[:120]}…")
    print()
